packages/cli/lib/lib/webpack/prerender.js

```diff
@@ -2,17 +2,18 @@ const { red, yellow } = require('kleur');
 const { resolve } = require('path');
 const { readFileSync } = require('fs');
 const stackTrace = require('stack-trace');
-const URL = require('url');
+const { URL } = require('url');
 const { SourceMapConsumer } = require('source-map');
+const prerender = require('./preact-iso-prerender');
 
-module.exports = function (env, params) {
+module.exports = async function (env, params) {
 	params = params || {};
 
 	let entry = resolve(env.dest, './ssr-build/ssr-bundle.js');
 	let url = params.url || '/';
 
 	global.history = {};
-	global.location = { ...URL.parse(url) };
+	global.location = new URL(url, 'http://localhost');
```

Comment:
Note: Moving to the WHATWG URL API is probably a breaking change. This does rather notably set things like `location.hostname` where previously they were null. Hopefully no one is relying on that to determine ssr vs browser envs.